### DISCUSSION TITLE
The first Map call is very slow

### DISCUSSION THREAD

**Discussion by zraffaele** [LuckyPennySoftware/AutoMapper#3753] [NA]
Hi,
i've create an ASP NET Core Web API with EF core, this have a lot of relationship within the tables
The automapper is configured with profile: CreateMap<ActivityBase, ActivityBaseModel>().ReverseMap().PreserveReferences();
In the startup register the Mapper register the DI is with services.AddAutoMapper(System.Reflection.Assembly.GetExecutingAssembly());
In the controller first execute the query and the result is mapped with this line of code:
return Mapper.Map<TModel>(result); The query return one record.
The first time that i execute this line of code is very very slow  10 second, at the second attemp the execution is instant.
Anyone please have any suggestions? 
Thanks in advance

RZ

**Comment by lbargaoanu** [comment]
https://github.com/AutoMapper/AutoMapper/blob/upgrade_guide/docs/Configuration.md#long-compilation-times
Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build. Set `MaxExecutionPlanDepth` to zero. Remove `MaxDepth` and `PreserveReferences`.

**Reply by lbargaoanu** [reply]
@zraffaele Thank you for taking the time to make the repro, apparently there was a bug with `MaxExecutionPlanDepth`, but as far I can tell, it's solved now.

**Comment by Erythnul** [comment]
Also potentially a combination effect with EF core model compilation, which can also take quite a bit of time. Unless your mapping is extremely complex this might be a more likely culprit.
FYI, EF core 6 intends to fix this by allowing precompiled models:
https://devblogs.microsoft.com/dotnet/announcing-entity-framework-core-6-0-preview-5-compiled-models/

**Reply by zraffaele** [reply]
When I call the EF mapping it has already returned the result and loaded all the data into the objects.
